autoload/codefmt/clangformat.vim

```diff
@@ -171,7 +171,7 @@ function! codefmt#clangformat#GetFormatter() abort
     let l:result = maktaba#syscall#Create(l:cmd).WithStdin(l:input).Call()
     let l:formatted = split(l:result.stdout, "\n")
 
-    if l:supports_cursor
+    if l:supports_cursor && !(line('$') == 1 && getline(1) == '')
```

Comment:
Would it make sense to do the empty buffer check above and avoid sending `-cursor` at all? That way it's not depending on subtle behaviors that might change.

Reply:
Ideally I'd prefer to bail out of the function as early as possible if the buffer is empty - really there's nothing to do.

Unfortunately there's a few other tests that rely on `clang-format` getting called no matter what when `FormatCode` is called.

I considered fixing the tests, but that was expanding the scope a bit too much for the amount of time I was willing to invest.

I could open a bug if you'd like so that at least we can keep track of it.

I guess another option is to call `clang-format` at the beginning of the function and then bail out if the buffer is empty.

Reply:
Right, skipping the whole formatting call would do similar, but would break more tests and intuitive assumptions. My main concern is relying on subtle (undocumented?) behavior that could conceivably change out from under us and have really strange results, if clang-format did someday decide to start returning the requested cursor coords in that case for consistency and we misparsed them.

If it's not too many affected tests and we can fix them in this PR to not have empty buffers, that'd be ideal (I don't think it's _that_ many tests that explicitly check for the `-cursor` option?). Otherwise, a tracking bug is better than nothing.

Reply:
I'll take a look, you're probably right that it's probably not _that_ many tests.

Reply:
😅

turns out that moving the check to the beginning doesn't affect _any_ other tests. I don't know why I thought it did - I thought before when I tried it, I saw some broken tests. but I just ran `vroom vroom/` and didn't get any failures.

marking this as resolved.